openthread-tock/create_openthread-tock.sh

```diff
@@ -0,0 +1,79 @@
+#! /bin/bash
+
+
+normal=$(tput sgr0)
+red=$(tput setaf 1)
+green=$(tput setaf 2)
+
+# This script expects to execut in the libtock-c/openthread-tock directory. Confirm this is true.
+current_directory=$(basename "$(pwd)")
+if [ "$current_directory" != "openthread-tock" ]; then
+    echo "${red}Error: The create_openthread-tock script must be invoked\
+	    in the 'openthread-tock' directory. Exiting.${normal}"
+    exit
+fi
+
+
+# Confirm that the git submodule has been checked out.
+git submodule status -- openthread >/dev/null 2>&1
+
+if [ $? -ne 0 ]; then
+    echo "${red}Error: git submodule 'openthread' not found.${normal}" 
+    echo "To checkout the submodule, use the command ${green}$ git submodule update --init -- openthread${normal}"
+    exit 
+fi
```

Comment:
If we're going to use a submodule this needs to "just work", aka do this for me.

Reply:
No. Let's just use a .zip. Then we also get the benefit of working if someone isn't using git and just downloads the code.

Reply:
I disagree. This is distinct from how we distribute toolchains, where ZIPs are arguably the best solution. Here we just want to just some pre-existing codebase, and with a ZIP we lose all of the benefits of using the code directly, such as the connection to a specific git commit, being able to immediately tell the authenticity of the code we pull in, etc. The recent attack on XZ arguably has shown us the importance on reducing reliance on non-auditable blobs.

Reply:
We also get all of the downsides of git submodules, too. The clunky interface, the challenges when developing on multiple branches, the inability to have multiple versions concurrently, the frustration with even discovering you _need_ a submodule, the conflicts if you change something in the submodule and the referenced commit changes, the inability to just download the library.

I agree that a zip makes it harder to tell if the zip is from the intended source. I'm not particularly concerned with this compared to ease of use. I disagree with the other two points (we use zips for libraries, not toolchains, and the commit is _glaringly_ obvious (much more so than with git and its hidden files). What are the others (etc)?

Reply:
I'm less and less convinced we can have users compile this when they build an openthread app. I think we need to provide precompiled libraries ala newlib. If we do that, that largely makes this issue moot.